src/mlpack/core/data/load_impl.hpp

```diff
@@ -156,16 +155,18 @@ bool LoadMatrix(const std::string& filename,
   return success;
 }
 
-template<typename MatType, typename DataOptionsType>
+template<typename ObjectType, typename DataOptionsType>
 bool Load(const std::string& filename,
-          MatType& matrix,
+          ObjectType& matrix,
           DataOptionsType& opts,
-          std::enable_if_t<IsArma<MatType>::value ||
-              IsSparseMat<MatType>::value>*,
-          std::enable_if_t<!std::is_same_v<DataOptionsType, bool>>*)
+          const typename std::enable_if_t<
+              IsDataOptions<DataOptionsType>::value>*)
 {
   Timer::Start("loading_data");
-
+  static_assert(!IsArma<ObjectType>::value || !IsSparseMat<ObjectType>::value
+      || !HasSerialize<ObjectType>::value, "mlpack can load Armadillo,"
+      " or a serialized mlpack model, please use a known type or"
+      " provide specific overloads.");
```

Comment:
```suggestion
      || !HasSerialize<ObjectType>::value, "mlpack can load Armadillo"
      " matrices or serialized mlpack models only; please use a known type.");
```

'provide specific overloads' is not something a user will know how to do; including that bit can only confuse them.  I know what you mean, which is that support should be added to the library or the class being loaded, but we can't expect users to do that.  Anyone who understands what 'provide specific overloads' means will also easily understand that their class is missing `serialize()` or similar.